includes/polling/os/ciscosb.inc.php

```diff
@@ -21,15 +21,17 @@
 if (! $hwversion) {
     $hwversion = $data['1']['rlPhdUnitGenParamHardwareVersion'];
 }
-if ($device['sysObjectID'] == '.1.3.6.1.4.1.9.6.1.89.26.1') {
+
+if (preg_match('/\.1\.3\.6\.1\.4\.1\.9\.6\.1\.72\.(....).+/', $device['sysObjectID'], $model)) {
```

Comment:
`(....).+`  is the same as `.{5,}`
would `starts_with($device['sysObjectID'], '.1.3.6.1.4.1.9.6.1.72.')` be appropriate?

Reply:
I need the capture group, because model is in sysObjectID.
And number of ports is in sysDescr.
I didn't find model in any 'easy' part in the snmp data.

seti# grep SGE /tmp/sge.txt
.1.3.6.1.2.1.47.1.1.1.1.2.67108992 = STRING: "SGE2010"
.1.3.6.1.2.1.47.1.1.1.1.2.68420352 = STRING: "SGE2010"
.1.3.6.1.2.1.47.1.1.1.1.2.134217856 = STRING: "SGE2010"
.1.3.6.1.2.1.47.1.1.1.1.2.201326720 = STRING: "SGE2010"
.1.3.6.1.2.1.47.1.1.1.1.2.268435584 = STRING: "SGE2010"
.1.3.6.1.2.1.47.1.1.1.1.2.335544448 = STRING: "SGE2010"
.1.3.6.1.2.1.47.1.1.1.1.2.402653312 = STRING: "SGE2010"
.1.3.6.1.2.1.47.1.1.1.1.2.469762176 = STRING: "SGE2010"
.1.3.6.1.2.1.47.1.1.1.1.2.536871040 = STRING: "SGE2010"
.1.3.6.1.2.1.47.1.1.1.1.7.68420352 = STRING: "SGE2010"

For an SGE2010 with 48 ports no POE:
SNMPv2-MIB::sysDescr.0 = STRING: 48-port 10/100/1000 Ethernet Switch
SNMPv2-MIB::sysObjectID.0 = OID: SNMPv2-SMI::enterprises.9.6.1.72.2010.2

#### Load poller module os ####
SNMP['/usr/bin/snmpget' '-v3' '-l' 'authNoPriv' '-n' "" '-a' 'SHA' '-A' 'PASSWORD' '-u' 'USER' '-OQUs' '-m' 'CISCOSB-DEVICEPARAMS-MIB:CISCOSB-Physicaldescription-MIB' '-M' '/disco2/opt/librenms/mibs:/disco2/opt/librenms/mibs/cisco' 'udp:HOSTNAME:161' 'genGroupHWVersion.0' 'rlPhdUnitGenParamModelName.1' 'rlPhdUnitGenParamHardwareVersion.1' 'rlPhdUnitGenParamSerialNum.1' 'rlPhdUnitGenParamSoftwareVersion.1' 'rlPhdUnitGenParamFirmwareVersion.1']
genGroupHWVersion.0 = 00.00.01
rlPhdUnitGenParamModelName.1 = No Such Object available on this agent at this OID
rlPhdUnitGenParamHardwareVersion.1 = 00.00.01
rlPhdUnitGenParamSerialNum.1 = DNI1446B72Z
rlPhdUnitGenParamSoftwareVersion.1 = *
rlPhdUnitGenParamFirmwareVersion.1 = *

Hardware: SGE2010-48 00.00.01
Version: Software 3.0.0.18, Firmware 2.0.0.03
Features:
Serial: DNI1446B72Z